clash-lib/src/Clash/Backend/Verilog.hs

```diff
@@ -1227,6 +1227,11 @@ exprLit k (Just (hty,sz)) (NumLit i) = case hty of
   _ -> int sz <> string "'b" <> blit
   where
     blit = bits k (toBits sz i)
+    i = case hty of
+             Signed _ -> let mask = 2^(sz-1) in case divMod i0 mask of
+                (s,i'') | even s    -> i''
+                        | otherwise -> i'' - mask
+             _ -> i0 `mod` 2^sz
```

Comment:
I'm not sure what's happening here. Could you add a (brief) comment explaining what this does?

Reply:
I just copied it from the [VHDL backend](https://github.com/clash-lang/clash-compiler/blob/b9212126339fe3a587c4f4d701bb7a87e73ef3a7/clash-lib/src/Clash/Backend/VHDL.hs#L1882-L1885)